README.md

```diff
@@ -1,53 +1,153 @@
+
+For the machine learning process, the code can be found under `/ml`. Three different classification techniques are applied on the data: Naive-Bayes, Random Forest and Support Vector Machines. For each of these techniques, three different data processing methods are used to numerically represent our data through vectors: word frequencies, TFIDF and Doc2Vec. The usage of word embeddings is important for this project because the importance of each word and their semantics in a text encapsulate information that may increase the performance of our models. Feature scaling was also applied to scale the vector range into [0,1] to satisfy the requirements for the algorithms, namely Naive-Bayes. The default ratio for training and test sets to be used is of 80-20 respectively.
+
+To evaluate the results of the trained models, the F1 score, accuracy, precision and recall of the predictions are used as performance measures.
+
+## III. Results
+
+### III.a. A First Look at the Data
+
+Before diving into the results obtained from applying machine learning classification techniques, we have chosen to take a first look at the data. Scripts for the generation of the following section's data can be found under `/data_exploration`.
+
+The next three figures illustrate the statistical data obtained from observing the distribution of labels across the posts, the distribution of posts across week days, and the distributions of top words across different labels.
+
+<p align="center">
+    <img src="data_exploration/imgs/class_distribution.png" width=375> </br>
+    Figure 1: Distribution of Labels for AITA Posts
+</p>
+
+From figure 1, the distribution tells us that the real data is greatly imbalanced with the majority of posts labelled as NTA. This observation has pushed us to later randomly undersample the posts labelled as NTA as to remove bias towards the majority class.
+
+<p align="center">
+    <img src="data_exploration/imgs/time_analysis.png" width=375> </br>
+    Figure 2: Distribution of AITA Posts by Week Days
+</p>
+
+<p align="center">
+    <img src="data_exploration/imgs/word_frequency.png" width=500> </br>
+    Figure 3: Distributions of the Top 30 Words by Label
+</p>
+
+We also attempted to extract topics from the content of the posts. Labelling for the topics is mostly difficult due to the sheer amount of words involved and the few number of topics chosen to group by for all the data. The following are a few interesting results from this process, tuples are displayed as (word, weight_of_word_for_topic):
```

Comment:
I would mention that it is us, and not the algorithm, which picked the topic names you mention